gateway-service-definitions/src/main/resources/services/atlas/2.1.0/rewrite.xml

```diff
@@ -32,14 +32,20 @@
         <rewrite template="{$frontend[path]}/atlas/"/>
     </rule>
 
-    <rule dir="OUT" name="ATLAS/atlas/outbound/login/headers/location">
-        <match pattern="*://*:*/login.jsp"/>
-        <rewrite template="{$frontend[url]}/atlas/login.jsp"/>
+    <rule dir="OUT" name="ATLAS/atlas/outbound/login/headers/location" flow="OR">
+        <match pattern="*://*:*/login.jsp">
+            <rewrite template="https://{$frontend[addr]}/gateway/knoxsso/knoxauth/logout.jsp?originalUrl={$postfix[url,/atlas/]}" />
+        </match>
+
+        <match pattern="*://*:*/logout.html?{**}">
+            <rewrite template="https://{$frontend[addr]}/gateway/knoxsso/knoxauth/logout.jsp?{**}"/>
```

Comment:
Here we are assuming that the cluster name is `gateway` which will not be the case always.

Reply:
@moresandeep Fix would be to replace gateway with {gateway} and knoxsso with {knoxsso}. Correct?

Reply:
We can leave knoxsso as is since it is not a custom topology name, for gateway I think we can use `{gateway.path}`